### PR TITLE
Misc changes

### PR DESCRIPTION
**Pull request by fingolfin**
- **Remove redundant gcdinv methods**
- **Enhance fpPolyRingElem constructor to accept more Integer types**
- **Fix missing return values in two unsafe ops**

### CODE REVIEW

File: src/flint/fmpq.jl [hunk 1]
```diff
@@ -369,6 +369,16 @@ function isless(a::QQFieldElem, b::QQFieldElem)
   return cmp(a, b) < 0
 end
 
+function Base.isapprox(x::QQFieldElem, y::QQFieldElem;
+                       atol::Real=0, rtol::Real=0,
+                       nans::Bool=false, norm::Function=abs)
+  if norm === abs && atol < 1 && rtol == 0
```
Comment (lgoettgens) [inline src/flint/fmpq.jl:375]: I dont really understand the atol<1 part here. For ZZRingElem, it's clear that two values with abs tolerance <1 already have to be equal. But I don't see why we can reason the same for QQFieldElem

File: src/flint/fmpz.jl [hunk 2]
```diff
@@ -1425,15 +1435,6 @@ function gcdx(a::ZZRingElem, b::ZZRingElem)
...
-@doc raw"""
-    gcdinv(a::ZZRingElem, b::ZZRingElem)
-
-Return a tuple $g, s$ where $g$ is the greatest common divisor of $a$ and
-$b$ and where $s$ is the inverse of $a$ modulo $b$ if $g = 1$. This function
-can be used to detect impossible inverses, i.e. where $a$ and $b$ are not
-coprime, and to yield the common factor of $a$ and $b$ if they are not
-coprime. We require $b \geq a \geq 0$.
-"""
 function gcdinv(a::ZZRingElem, b::ZZRingElem)
```
Comment (lgoettgens) [inline src/flint/fmpz.jl:1438]: It seems you forgot to remove the method here? Or was this a genuine decision?

File: src/flint/fmpq.jl (hunk 1 above)
Comment (fingolfin) [reply src/flint/fmpq.jl]: This was indeed wrong. For now I removed the `isapprox` changes from this PR in the hope that the rest can be merged, I'll fix the `isapprox` in a separate PR.

(What happened: I first copied the Julia library method for `Integer` and adjusted that. Then later I discovered I also needed it for `QQFieldElem` and just copied that without thinking. It was late *sigh*. Anyway, fixed version will come later)

File: src/flint/fmpz.jl (hunk 2 above)
Comment (fingolfin) [reply src/flint/fmpz.jl:1428]: This method calls a dedicated C function `fmpz_gcdinv` and hence was not removed.

**Review by lgoettgens — APPROVED** [review]

**Pull request merged**